grr/server/grr_response_server/gui/api_plugins/metadata_test.py

```diff
@@ -0,0 +1,382 @@
+    expected["field_int32"] = expected_int32
+
+    expected_fixed64 = {"type": "integer", "format": "uint64"}
+    expected["field_fixed64"] = expected_fixed64
+
+    expected_fixed32 = {"type": "integer", "format": "uint32"}
+    expected["field_fixed32"] = expected_fixed32
+
+    expected_bool = {"type": "boolean"}
+    expected["field_bool"] = expected_bool
+
+    expected_string = {"type": "string"}
+    expected["field_string"] = expected_string
+
+    expected_bytes = {"type": "string", "format": "binary"}
+    expected["field_bytes"] = expected_bytes
+
+    expected_uint32 = {"type": "integer", "format": "uint32"}
+    expected["field_uint32"] = expected_uint32
+
+    expected_sfixed32 = {"type": "integer", "format": "int32"}
+    expected["field_sfixed32"] = expected_sfixed32
+
+    expected_sfixed64 = {"type": "integer", "format": "int64"}
+    expected["field_sfixed64"] = expected_sfixed64
+
+    expected_sint32 = {"type": "integer", "format": "int32"}
+    expected["field_sint32"] = expected_sint32
+
+    expected_sint64 = {"type": "integer", "format": "int64"}
+    expected["field_sint64"] = expected_sint64
+
+    expected_binarystream = {"type": "string", "format": "binary"}
+    expected["BinaryStream"] = expected_binarystream
+
+    self.assertDictEqual(expected, openapi_primitives)
+
+  def testRepetitiveField(self):
+    # Test the GET route. The repeated field should be a query parameter.
+    params = (self.openapi_desc_dict.get("paths").get("/metadata_test/method4")
+              .get("get").get("parameters"))
+    self.assertEqual(len(params), 1)
+
+    param = params[0]
+    self.assertEqual(param["in"], "query")
+    schema = param["schema"]
+    self.assertEqual(schema["type"], "array")
+    self.assertEqual(schema["items"]["type"], "integer")
+    self.assertEqual(schema["items"]["format"], "int64")
+
+    # Test the POST route. The repeated field should be in the request body.
+    req_body = (self.openapi_desc_dict.get("paths")
+                .get("/metadata_test/method4").get("post").get("requestBody"))
+    req_schema = req_body["content"]["application/json"]["schema"]
+    self.assertEqual(req_schema["type"], "object")
+    schema = req_schema["properties"]["field_repeated"]
+    self.assertEqual(schema["type"], "array")
+    self.assertEqual(schema["items"]["type"], "integer")
+    self.assertEqual(schema["items"]["format"], "int64")
+
+  def testEnumField(self):
+    # Test the GET route. The enum field should be a query parameter.
+    params = (self.openapi_desc_dict.get("paths").get("/metadata_test/method5")
+              .get("get").get("parameters"))
+    self.assertEqual(len(params), 1)
+
+    param = params[0]
+    self.assertEqual(param["in"], "query")
+    ref_get = param["schema"]["$ref"]
+    ref_nodes = ref_get.split("/")
+    ref_nodes = ref_nodes[1:]
+    schema = self.openapi_desc_dict
+    for node in ref_nodes:
+      schema = schema[node]
```

Comment:
Here the testing logic is pretty complicated. For example, as a reader, I'm not sure I can assess whether why `ref_nodes = ref_nodes[1:]` is done. Would it be possible to simplify this by using predefined literals? - since we know which values to expect.

Reply:
Made the change and used predefined literals everywhere. Please let me know if they are not ok. Thank you!